src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,32 +74,250 @@ const NotebookLauncher = _.flow(
     title: _.get('notebookName'),
     showTabBar: false
   })
-)(({ queryParams = {}, notebookName, workspace, workspace: { accessLevel, canCompute }, cluster, refreshClusters, onRequesterPaysError }, ref) => {
-  return (Utils.canWrite(accessLevel) && canCompute && !queryParams['read-only']) ?
-    h(NotebookEditor, { notebookName, workspace, cluster, refreshClusters }) :
-    h(Fragment, [
-      h(ReadOnlyMessage, { notebookName, workspace }),
-      h(NotebookPreviewFrame, { notebookName, workspace, onRequesterPaysError })
+)(
+  ({ queryParams = {}, notebookName, workspace, workspace: { workspace: { namespace, name }, accessLevel, canCompute }, cluster, refreshClusters, onRequesterPaysError }, ref) => {
+    const [createOpen, setCreateOpen] = useState(false)
+    // Status note: undefined means still loading, null means no cluster
+    const clusterStatus = cluster && cluster.status
+
+    return h(Fragment, [
+      Utils.cond([
+        !(Utils.canWrite(accessLevel) && canCompute), () => h(Fragment, [
+          h(PreviewHeader, { cluster, refreshClusters, notebookName, workspace, readOnlyAccess: true }),
+          h(NotebookPreviewFrame, { notebookName, workspace })
+        ])
```

Comment:
This case looks very similar to the default. Can we combine the two, and just parameterize `readOnlyAccess`?